pandas/core/indexes/base.py

```diff
@@ -2110,7 +2110,7 @@ def argsort(self, *args, **kwargs):
         return result.argsort(*args, **kwargs)
 
     def __add__(self, other):
-        return Index(np.array(self) + other)
+        return Index(np.array(self) + other, name=self.name)
```

Comment:
actually this is not correct. We don't assign the index name on a boolean operation. *only* on an inplace operation (e.g. ``__iadd__``). This also would apply to other ops like ``isub`` etc.